sdks/python/apache_beam/runners/interactive/caching/streaming_cache.py

```diff
@@ -166,13 +169,15 @@ def _wait_until_file_exists(self, timeout_secs=30):
 
     # Wait for up to `timeout_secs` for the file to be available.
     start = time.time()
-    path = os.path.join(self._cache_dir, *self._labels)
-    while not os.path.exists(path):
+    while not os.path.exists(self._path):
       time.sleep(1)
       if time.time() - start > timeout_timestamp_secs:
+        from apache_beam.runners.interactive.pipeline_instrument import CacheKey
+        pcollection_var = CacheKey.from_str(self._labels[-1]).var
```

Comment:
Yeah, because these are Python variables, they have very strict naming conventions which is ok to use for file names (alphanumeric with the addition of an underscore).